.github/workflows/multi-arch-build.yaml

```diff
@@ -62,26 +62,37 @@ jobs:
           echo "$VERSION_OUTPUT"
           echo ::set-output name=version_output::$VERSION_OUTPUT
 
-      # Generate image related info - names, labels
+      # Generate image related info - names, labels, check whether to push
       - name: Generate image information
         id: image_info
         run: |
-          # have special environment variable for image name.
+          # Have special environment variable for image name.
           # Names are different for upstream and stable Buildah images
           if [ "${{ matrix.source }}" == "upstream" ]; then
             # quay.io/buildah/upstream:master
-            echo ::set-output name=buildah_tag::"${{ env.BUILDAH_QUAY_REGISTRY}}/${{ matrix.source }}:master"
+            # Always push upstream image to buildah repository
+            echo ::set-output name=buildah_push::'true'
+            # Never push upstream image to containers repository
+            echo ::set-output name=containers_push::'false'
+            echo ::set-output name=buildah_tag::"${{ env.BUILDAH_QUAY_REGISTRY}}/${{ matrix.source }}:main"
           fi
 
           if [ "${{ matrix.source }}" == "stable" ]; then
             VERSION=v"$(echo "${{ steps.sniff_test.outputs.version_output }}" | head -n 1 | awk '{print $2}')"
             # quay.io/buildah/stable:vX.X.X
+            # Check if stable image with current version already exists and make decision if new one needs to be pushed
+            PUSH=$(docker pull ${{ env.BUILDAH_QUAY_REGISTRY}}/${{ matrix.source }}:${VERSION} &>/dev/null && echo 'false' || echo 'true')
```

Comment:
I think not, but any chance to use Podman or Buildah here instead of Docker?  @cevich?

Reply:
I guess `docker` can be switched to `podman` here, however I don't see much advantage in this replacement. If there is more elegant way to check existence of remote container image with `skopeo` or `podman`, I'd be happy to use it, but I need your input here as experts :)

Reply:
You should be able to use `skopeo list-tags docker://${{ env.BUILDAH_QUAY_REGISTRY}}/${{ matrix.source }} | jq -r .Tags[] | grep -q "$VERSION."` for this, it will be _MUCH_ faster as well.